packages/planner/src/app/components_ngrx/export-csv/export-csv.component.ts

```diff
@@ -0,0 +1,44 @@
+import { Component, Input, Inject } from '@angular/core';
+import { FilterService } from '../../services/filter.service';
+import { WIT_API_URL, Space } from 'ngx-fabric8-wit';
+import { SpaceQuery } from '../../models/space';
+import { Observable, of, combineLatest } from 'rxjs';
+import { filter, switchMap } from 'rxjs/operators';
+import { EQUAL, AND } from '../../services/query-keys';
+import { ActivatedRoute } from '@angular/router';
+
+@Component({
+  selector: 'export-csv',
+  templateUrl: './export-csv.component.html',
+})
+export class ExportCsvComponent {
+  @Input() isDisabled: boolean;
+  @Input() query: string;
+
+  public csvLink: Observable<string> = combineLatest(
+    this.spaceQuery.getCurrentSpace.pipe(filter((s) => !!s)),
+    this.route.queryParams.pipe(filter((q) => !!Object.keys(q).length)),
+  ).pipe(
+    switchMap(([space, query]) => {
+      const currentQuery = this.filterService.queryToJson(query.q);
+      const spaceQuery = this.filterService.queryBuilder('space', EQUAL, space.id);
+      const expression = {
+        expression: this.filterService.queryJoiner(currentQuery, AND, spaceQuery),
+      };
+      return of(
+        this.baseApiURL +
+          'search/workitems/csv?' +
+          Object.keys(expression)
+            .map((k) => 'filter[' + k + ']=' + encodeURIComponent(JSON.stringify(expression[k])))
+            .join('&'),
+      );
+    }),
+  );
+
+  constructor(
```

Comment:
The style / standard we follow has the constructor listed above the functions.

Reply:
That's an Observable being assigned to `csvLink` variable. should I move it below constructor?

Reply:
I'm not blocking the merge on this, it just looks odd to me. It is not just a variable name that is being set. You are doing something there and that usually happens below the constructor. This is for readability not functionality.

Reply:
Done. Listed the `csvLink` below constructor.